### PR TITLE
Allow dynamic adding of items

### PR DESCRIPTION
**Pull request by midzer**
Hi,
pretty large PR for this feature. Feedback and review is very welcome.

Have a nice week
midzer

### CODE REVIEW

File: js/tobi.js [hunk 1]
```diff
@@ -716,10 +716,15 @@
 
       // Execute a few things once per element
       [].forEach.call(elements, function (element) {
-        initElement(element);
+        initElement(element)
       })
     }
 
+    // Make private function available from outside
+    this.add = function(element) {
```
Comment (rqrauhvmra) [inline js/tobi.js]: This do not work.

> Cannot set property 'add' of undefined

Maybe I must rewrite the code in JavaScript constructor functions. Have you a working example?

File: js/tobi.js [hunk 2]
```diff
@@ -669,26 +666,9 @@
...
-      // Get a list of all elements within the document
-      var elements = document.querySelectorAll(config.selector)
-
-      if (!elements.length) {
-        throw new Error('Ups, I can\'t find the selector ' + config.selector + '.')
-      }
-
-      // Execute a few things once per element
-      [].forEach.call(elements, function (element, index) {
+    this.initElement = function initElement(element) {
```
Comment (rqrauhvmra) [inline js/tobi.js]: Missing space before function parentheses.

File: js/tobi.js [hunk 3]
```diff
@@ -669,26 +666,9 @@
...
+      if (gallery.indexOf(element) === -1) {
+        gallery.push(element);
```
Comment (rqrauhvmra) [inline js/tobi.js]: Unnecessary semicolon.

File: js/tobi.js [hunk 4]
```diff
@@ -706,21 +686,41 @@
...
+      transformProperty = transformSupport()
+
+      // Get a list of all elements within the document
+      var elements = document.querySelectorAll(config.selector)
 
-      galleryLength = gallery.length
-      createOverlay()
+      if (!elements.length) {
+        console.log('Ups, I can\'t find the selector ' + config.selector + '.')
+        return;
```
Comment (rqrauhvmra) [inline js/tobi.js]: Unnecessary semicolon.

File: js/tobi.js [hunk 5]
```diff
@@ -21,6 +21,7 @@
     root.Tobi = factory()
   }
 }(this, function () {
+  'use strict';
```
Comment (rqrauhvmra) [inline js/tobi.js]: Unnecessary semicolon.

File: js/tobi.js [hunk 6]
```diff
@@ -667,7 +667,7 @@
       document.removeEventListener('focus', trapFocus)
     }
 
-    this.initElement = function initElement(element) {
+    var initElement = function initElement(element) {
```
Comment (rqrauhvmra) [inline js/tobi.js]: Missing space before function parentheses.

File: js/tobi.js (hunk 1 above)
Comment (midzer) [reply js/tobi.js]: Yes, now there is a `new` constructor necessary. I did not find another way around. See https://github.com/midzer/eisolzried/blob/master/_assets/js/imagegal.js

I can update README.md afterwards as well.

Comment (rqrauhvmra) [reply js/tobi.js]: Ah, ok :)

I included your fork [here](https://codepen.io/rqrauhvmra/pen/ZRZdwG), but how can I use `tobi.add()`?
If I try it in the console I receive:

> tobi is not defined

Comment (midzer) [reply js/tobi.js]: Your example is working fine for me after disabling adblocker.

For dynamic adding (no initial `.lightbox` class `<a>` linked images found) try something like
```
tobi.add(document.getElementById('1'))
```
for a referenced `id="1"` `<a>`

Comment (rqrauhvmra) [reply js/tobi.js]: [I get it](https://codepen.io/rqrauhvmra/pen/ZRZdwG)  :)

Thanks!

**Comment by rqrauhvmra** [comment]
Thanks again for your PR @midzer 

If you like, you can update the README.md

**Pull request merged**